### PR TITLE
feat(surveys): wip add link survey ui

### PR DESCRIPTION
**Pull request by sifferhans**
## About the changes
I've implemented a new type of survey; links.
This can be used to send the user to external websites, eg. for scheduling interviews.

### Important files
- `lib/components/surveys/survey_link.dart`
- `lib/components/surveys/bottom_sheet_survey.dart`

https://github.com/user-attachments/assets/96a6565e-2419-4099-ac8d-3669cd06a8ee

### CODE REVIEW

File: lib/components/surveys/survey_link.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            height: 116,
+            margin: const EdgeInsets.only(top: 16),
+            child: Column(
+              children: [
+                SizedBox(
+                  width: double.infinity,
+                  child: design.buttons.large(
+                    labelText: actionButtonText,
+                    onPressed: () async {
+                      final uri = Uri.parse(url);
```
Comment (KillerX) [inline lib/components/surveys/survey_link.dart:69]: Should this parsing maybe happen earlier so we don't crash "late" on an invalid URL?

File: lib/components/surveys/bottom_sheet_survey.dart [hunk 2]
```diff
@@ -67,11 +68,12 @@ class BottomSheetSurvey extends HookWidget {
 
     return PopScope(
       canPop: false,
-      onPopInvoked: (v) async {
+      onPopInvokedWithResult: (v, r) {
```
Comment (KillerX) [inline lib/components/surveys/bottom_sheet_survey.dart:71]: What are the consequences of this not being `async` anymore?

File: lib/components/surveys/survey_link.dart [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          Container(
+            height: 116,
+            margin: const EdgeInsets.only(top: 16),
+            child: Column(
+              children: [
+                SizedBox(
+                  width: double.infinity,
+                  child: design.buttons.large(
+                    labelText: actionButtonText,
+                    onPressed: () async {
```
Comment (KillerX) [inline lib/components/surveys/survey_link.dart:68]: Also what happens with the popup once the user returns?

**Review by KillerX — DISMISSED** [review]
Just a few questions that are probably good to think through if they have not been considered yet, but looks ok to me otherwise.

File: lib/components/surveys/survey_link.dart (hunk 3 above)
Comment (sifferhans) [reply lib/components/surveys/survey_link.dart]: The popup will now close

**Review by KillerX — APPROVED** [review]

**Pull request merged**